controllers/profiles/unfollowUserV3.js

```diff
@@ -38,38 +40,86 @@ module.exports = async (req, res) => {
+    const selfAnonymousUser = await UsersFunction.findAnonymousUserId(User, req?.userId);
+
+    // unfollow_process
+    // 1 main_feed {sign_follower_user_id} user_excl {sign_followed_user_id}
+    // 2 main_feed {sign_follower_user_id} user {sign_followed_user_id}
+    // 3 main_feed_following {sign_follower_user_id} user_excl {sign_followed_user_id}
+    // 4 main_feed_following {sign_follower_user_id} user {sign_followed_user_id}
+    // 5 main_feed_following {sign_follower_user_id} user_anon {anon_followed_user_id}
+    // 6 main_feed {sign_follower_user_id} user_anon {anon_followed_user_id}
+    // 7 main_feed {sign_followed_user_id} user_anon {anon_follower_user_id}
+
+    const unfollow_process_1 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_NAME,
       req?.userId,
+      GetstreamConstant.USER_EXCLUSIVE_FEED_NAME,
       user_id_followed
     );
-    const sendJobUnfollowF2User = sendUnFollowMainFeedF2(req.userId, user_id_followed);
-    await Promise.all([
-      unfollowUserExcl,
-      unfollowUser,
-      unfollowMainFeedFollowing,
-      sendJobUnfollowF2User
-    ]);
 
-    const anonymousUser = await UsersFunction.findAnonymousUserId(User, user_id_followed);
-    const selfAnonymousUser = await UsersFunction.findAnonymousUserId(User, req?.userId);
-    await Getstream.feed.unfollowAnonUser(
-      req?.token,
+    const unfollow_process_2 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_NAME,
       req?.userId,
-      user_id_followed,
-      selfAnonymousUser?.user_id,
+      GetstreamConstant.USER_FEED_NAME,
+      user_id_followed
+    );
+
+    const unfollow_process_3 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_FOLLOWING_NAME,
+      req?.userId,
+      GetstreamConstant.USER_EXCLUSIVE_FEED_NAME,
+      user_id_followed
+    );
+
+    const unfollow_process_4 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_FOLLOWING_NAME,
+      req?.userId,
+      GetstreamConstant.USER_FEED_NAME,
+      user_id_followed
+    );
+
+    const unfollow_process_5 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_FOLLOWING_NAME,
+      req?.userId,
+      GetstreamConstant.USER_ANON_FEED_NAME,
+      anonymousUser?.user_id
+    );
+
+    const unfollow_process_6 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_NAME,
+      req?.userId,
+      GetstreamConstant.USER_ANON_FEED_NAME,
       anonymousUser?.user_id
     );
+
+    const unfollow_process_7 = unfollowFeed(
+      GetstreamConstant.MAIN_FEED_NAME,
+      user_id_followed,
+      GetstreamConstant.USER_ANON_FEED_NAME,
+      selfAnonymousUser?.user_id
+    );
+
+    const sendJobUnfollowF2User = sendUnFollowMainFeedF2(req.userId, user_id_followed);
+
+    await Promise.all([
+      unfollow_process_1,
+      unfollow_process_2,
+      unfollow_process_3,
+      unfollow_process_4,
+      unfollow_process_5,
+      unfollow_process_6,
+      unfollow_process_7,
+      sendJobUnfollowF2User
+    ]);
   } catch (e) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is repetitive and could be simplified by creating a helper function that takes the necessary parameters and returns the result of `unfollowFeed`. This would make the code more DRY (Don't Repeat Yourself) and easier to maintain.

Here's an example of how you could refactor this:

```javascript
const createUnfollowProcess = (feedName, userId, targetFeed, targetUserId) => {
  return unfollowFeed(feedName, userId, targetFeed, targetUserId);
};

const unfollowProcesses = [
  createUnfollowProcess(GetstreamConstant.MAIN_FEED_NAME, req?.userId, GetstreamConstant.USER_EXCLUSIVE_FEED_NAME, user_id_followed),
  // ... add other processes here ...
];

await Promise.all([...unfollowProcesses, sendJobUnfollowF2User]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->